pkg/harvester/integrity/repository/disk.go

```diff
@@ -1 +1,237 @@
 package repository
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	//"geninterface/repository"
+	"io/ioutil"
+	"math/big"
+	"net/url"
+	"time"
+
+	"github.com/andres-erbsen/clock"
+)
+
+// Disk is a SigningCA that generates signing certificates and validation materials
+// Disk is a local signing CA and has access to root keys and root certs via a path.
+type Disk struct {
+	rootSigner       crypto.Signer       // Root Key
+	rootCertificate  *x509.Certificate   // Root Certificate CA
+	validationBundle []*x509.Certificate // List of certificates to be used for validation of signing certificates
+}
+
+type Config struct {
+	CertFilePath           string `hcl:"cert_file_path"`           // Root CA certificate
```

Comment:
Pattern for these comments (mostly for exported types) is that they should be on top of the declaration and start with the name of the defined variable. This is a standard Go "lightly" enforces and most IDEs leverage that to provide a better way to navigate through the code. 

```
// CertFilePath is the path for Root CA certificate
CertFilePath           string `hcl:"cert_file_path"`